src/main/java/org/spongepowered/api/context/ContextViewer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.context;
+
+import org.spongepowered.api.data.context.DataContext;
+import org.spongepowered.api.data.context.DataContextual;
+
+/**
+ * A {@link ContextViewer} has the ability to have its own unique "view"
+ * of data on applicable objects.
+ *
+ * <p>In the case of data, a {@link ContextViewer} has its own unique
+ * {@link DataContext view} of {@link DataContextual} data.</p>
+ */
+public interface ContextViewer {
```

Comment:
@gabizou what do you mean?

`ContextViewer` is a marker to not allow any random object to be passed places, as well as to check before casting.
Nothing is stored using `String` and I intend to keep it that way - returning a `NamedContextual` doesn't even make sense here